reframe/frontend/cli.py

```diff
@@ -367,6 +370,9 @@ def main():
 
         # Act on checks
 
+        if options.module_map:
+            get_modules_system().load_mapping_from_file(options.module_map[0])
+
```

Comment:
This should move up. Here is the part where we "act on checks". The treatment of this option should be at the beginning after the current system is set up.